SimCore/src/SimCore/Reweight/GenieReweightProducer.cxx

```diff
@@ -0,0 +1,203 @@
+//
+// Created by Wesley Ketchum on 4/29/24.
+//
+
+#include "SimCore/Reweight/GenieReweightProducer.h"
+
+#include <iostream>
+#include <random>
+
+#include "Framework/EventGen/EventRecord.h"
+#include "Framework/EventGen/HepMC3Converter.h"
+#include "Framework/Interaction/Interaction.h"
+#include "Framework/Utils/RunOpt.h"
+#include "HepMC3/GenEvent.h"
+#include "HepMC3/Print.h"
+#include "RwCalculators/GReWeightAGKY.h"
+#include "RwCalculators/GReWeightDISNuclMod.h"
+#include "RwCalculators/GReWeightDeltaradAngle.h"
+#include "RwCalculators/GReWeightFGM.h"
+#include "RwCalculators/GReWeightFZone.h"
+#include "RwCalculators/GReWeightINuke.h"
+#include "RwCalculators/GReWeightINukeParams.h"
+#include "RwCalculators/GReWeightNonResonanceBkg.h"
+#include "RwCalculators/GReWeightNuXSecCCQE.h"
+#include "RwCalculators/GReWeightNuXSecCCQEaxial.h"
+#include "RwCalculators/GReWeightNuXSecCCQEvec.h"
+#include "RwCalculators/GReWeightNuXSecCCRES.h"
+#include "RwCalculators/GReWeightNuXSecCOH.h"
+#include "RwCalculators/GReWeightNuXSecDIS.h"
+#include "RwCalculators/GReWeightNuXSecNC.h"
+#include "RwCalculators/GReWeightNuXSecNCEL.h"
+#include "RwCalculators/GReWeightNuXSecNCRES.h"
+#include "RwCalculators/GReWeightResonanceDecay.h"
+#include "RwCalculators/GReWeightXSecEmpiricalMEC.h"
+#include "RwCalculators/GReWeightXSecMEC.h"
+#include "RwFramework/GReWeight.h"
+#include "RwFramework/GReWeightI.h"
+#include "RwFramework/GSyst.h"
+#include "RwFramework/GSystSet.h"
+#include "SimCore/Event/EventWeights.h"
+#include "SimCore/Event/HepMC3GenEvent.h"
+
+namespace simcore {
+
+GenieReweightProducer::GenieReweightProducer(const std::string& name,
+                                             framework::Process& process)
+    : Producer(name, process) {
+  hepMC3Converter_ = new genie::HepMC3Converter;
+  genie_rw_ = new genie::rew::GReWeight;
+}
+
+GenieReweightProducer::~GenieReweightProducer() {
+  delete hepMC3Converter_;
+  delete genie_rw_;
+}
+
+void GenieReweightProducer::configure(framework::config::Parameters& ps) {
+  hepmc3CollName_ = ps.getParameter<std::string>("hepmc3CollName");
```

Comment:
Fun fact, in a Framework update a while ago, I changed `Parameters` to have a shorter accessor function.

Not really necessary, just using this as a soap-box to try to advertise this update a bit more.

```suggestion
  hepmc3CollName_ = ps.get<std::string>("hepmc3CollName");
```

Reply:
Updated!